src/components/combos/PostCombo.js

```diff
@@ -144,7 +144,21 @@ const PostCombo = ({ userCombo, setUserCombo }) => {
               type="number"
               value={yearEstablished}
               placeholder={new Date().getFullYear()}
-              onChange={(e) => setYearEstablished(e.target.value)}
+              onChange={(e) => {
```

Comment:
I created a method `handleYearChange` containing the logic. This makes it much more neat to look at, you are right!  
Also I got rid of the `onBlur` event since the same behavior can be achieved with only the `onChange` event.  
98bbfea29c876122fcea1a59dadb0c3d2f389e5b

Reply:
I also noticed another year-input field in PostTrick... So I have the function `handleYearChange` two times in two different files.  
In my opinion it isn't that bad, but do you know a more elegant way to go about this?

Reply:
No, sadly I don't. A helper / utilities script for "clamping" the year could be created and then referenced by both the Post and the Combo script but I think that would be even less elegant than writing the logic twice and it would create an unnecessary dependency. The function is so short, I think writing it twice is fine